package NYT.NChunkServer.NProto;

import "yt/ytlib/chunk_server/chunk_service.proto";
import "yt/ytlib/misc/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TMsgRegisterHolder
{
    required bytes address = 1;
    required NYT.NProto.TGuid incarnation_id = 2;
    required THolderStatistics statistics = 3;
}

message TMsgUnregisterHolder
{
    required int32 holder_id = 1;
}

// Full heartbeat changes are typcially large and as passed as-is in
// form of RPC requests. Hence we don't have any special
// TMsgFullHeartbeat type here.

message TMsgIncrementalHeartbeat
{
    required int32 holder_id = 1;
    required THolderStatistics statistics = 2;
    repeated TChunkAddInfo added_chunks = 3;
    repeated TChunkRemoveInfo removed_chunks = 4;
}

message TMsgUpdateJobs
{
    required int32 holder_id = 1;
    repeated TJobStartInfo started_jobs = 2;
    repeated TJobStopInfo stopped_jobs = 3;
}

////////////////////////////////////////////////////////////////////////////////

